parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -412,7 +412,16 @@ version_gte() {
 }
 
 systemctlEnableAndStart() {
-    systemctl_restart 100 5 30 $1
+    # Parameterize the timeout value here to assign longer timeout for mig-partition service to complete
+    # while keeping the original timeout value for other scenarios
+    service=$1
+    if [ "$service" = "mig-partition" ]; then
```

Comment:
I'd prefer bubbling the timeout parameter up to the top-level from systemctl_restart instead of needing to pivot based on service name

Reply:
refactored the implementation